roles/confluent.zookeeper/tasks/main.yml

```diff
@@ -14,6 +14,21 @@
     - ansible_os_family
     - ansible_fqdn
 
+- name: Get Package Facts
+  package_facts:
+    manager: auto
+  when: installation_method == "package"
+
+- name: Stop Zookeeper Service if upgrade
+  ansible.builtin.systemd:
+    name: "{{ zookeeper_service_name }}"
+    state: stopped
+  when:
+    - installation_method == "package"
+    - >
+      ansible_facts.packages[zookeeper_main_package] is not defined
```

Comment:
if the package is not installed is it necessary to stop the service?

Reply:
That is just in case the main package has changed name, but then it is special anyhow. I can remove it.